app/views/admin/solutions/show.html.haml

```diff
@@ -16,8 +16,8 @@
         .byline by #{display_handle_link(@solution.user, @solution_user_track)}
 
   .lo-container
-    .pure-g.panels
-      .pure-u-1.pure-u-md-1-2.lhs
+    = render "widgets/panels" do
```

Comment:
Rather than rendering a partial here, could you instead add a helper "render_panels" to the `PanelHelper` class you've created. You can then merge the `panel_classes_for` line into that.

Note also that `current_user` might be `nil`.

I think you the method you'll want is something like:
```ruby
def render_panels
  klass =  current_user.try(&:full_width_code_panes?) ? "panels--vertical-split" : ""
  content_tag :div, &block, class "panels #{klass}"
end
```